api/v1alpha1/releaseplan_webhook.go

```diff
@@ -0,0 +1,74 @@
+/*
```

Comment:
I think there should be a test for circular still.
```
if workspace == ""
target.namespace cannot be releaseplan.namespace

if workspace != ""
if target.namespace == releaseplan.namespace && target.workspace == releaseplan.labels["kcp-dev"] ; throw err
```
That is not the real label name but you get the idea

Reply:
I don't see any reason to add it. The reason why we did add it in the pass is that it could create a circle dependency in code (use the same object for both things). That cannot happen now.

Reply:
We only did it before because of a circular dependency in code? I thought we wanted to prevent users from targeting themselves on a conceptual level. If we don't care about the users doing that, sure we can leave it out but I didn't think that

Reply:
Yep, only circular dependency as you could reach to a ReleaseLink with no strategy.